src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4754,6 +4754,14 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                         end += (_info.Offset.Query - _info.Offset.Path);
                     }
                 }
+
+                // On Unix, escape '\\' in path of file uris to '%5C' canonical form.
+                if (!IsWindowsSystem && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes) && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && !IsImplicitFile)
```

Comment:
> the compiler will realize this is unreachable codepath
> will give warning, unreachable code detected

I smell bad code patterns here :

1. You are adding dead code for certain builds.
2. The compiler is giving you a warning that you are ignoring.
3. The code isn't readable by a human: it should be very clear that some code will actually not exist on certain platforms (why the #if statement exists and the reason for which we created the `*.Unix.cs` vs `.Windows.cs` files).
4. What does the code coverage tool tell you about the coverage of this cs file when running in Windows vs Unix? Will it understand the compiler decision or just report uncovered code (which can never be covered given the optimization).